components/reusable-components/reusable-form/ReusableForm.tsx

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-unused-vars */
+
+'use client';
+
+import React, { useState } from 'react';
+import * as Yup from 'yup';
+import { useFormik, FormikProps } from 'formik';
+import Link from 'next/link';
+import Image from 'next/image';
+import style from './reusableForm.module.scss';
+import TextInput from '../text-input/TextInput';
+import Button from '../button/Button';
+import github from '../../../public/icons-footer/github.svg';
+import linkedin from '../../../public/icons-footer/linkedIn.svg';
+import google from '../../../public/icons-footer/google.svg';
+import check from '../../../public/icons/check.svg';
+
+const ReusableForm = () => {
+  const [password, setPassword] = useState<string>('');
+
+  const initialValues = {
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+  };
+
+  const validationSchema = Yup.object({
+    firstName: Yup.string()
+      .matches(/^[a-zA-Z .'-]+$/, '*Невалидно име!')
+      .min(2, '*Вашето име е премногу кратко!')
+      .max(50, '*Вашето име е премногу долго!')
+      .required('*Задолжително внесете име'),
+    lastName: Yup.string()
+      .min(2, '*Вашето презиме е премногу кратко!')
+      .max(50, '*Вашето презиме е премногу долго!')
+      .required('*Задолжително внесете презиме'),
+    email: Yup.string()
+      .email('*Невалидна емаил адреса')
+      .required('*Задолжително внесете електронка пошта'),
+    password: Yup.string()
+      .matches(/[a-zA-Z]/, '*Лозинката не ги исполнува сите услови!')
+      .min(8, '*Вашата лозинка е премногу кратка!')
+      .required('*Задолжително внесете лозинка'),
+  });
+
+  const formik: FormikProps<{
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+  }> = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: () => {},
```

Comment:
Here on Submit we will update that state by testing each condition . Something like:
  ```
  onSubmit: (values) => {
      setPasswordValidation({
        uppercase: /[A-Z]/.test(values.password),
        specialChar: /[!@#$%^&*(),.?":{}|<>]/.test(values.password),
        minLength: values.password.length >= 8,
      });
    },
```